src/drawing/drawing_fast.cpp

```diff
@@ -0,0 +1,186 @@
+                    for (; no_pixels > 0; no_pixels -= zoom_amount, source_pointer += zoom_amount, dest_pointer++) {
+                        *dest_pointer = *source_pointer;
+                    }
+                }
+            }
+        }
+
+        //Add a line to the drawing surface pointer
+        next_dest_pointer += line_width;
+    }
+}
+
+#define DrawRLESpriteHelper2(image_type, zoom_level) \
+    DrawRLESprite2<image_type, zoom_level>(source_bits_pointer, dest_bits_pointer, palette_pointer, dpi, source_y_start, height, source_x_start, width)
+
+template<int image_type>
+static void FASTCALL DrawRLESprite1(const uint8* source_bits_pointer,
+                                      uint8* dest_bits_pointer,
+                                      const uint8* palette_pointer,
+                                      const rct_drawpixelinfo *dpi,
+                                      int source_y_start,
+                                      int height,
+                                      int source_x_start,
+                                      int width)
+{
+    int zoom_level = dpi->zoom_level;
+    switch (zoom_level) {
+    case 0: DrawRLESpriteHelper2(image_type, 0); break;
+    case 1: DrawRLESpriteHelper2(image_type, 1); break;
+    case 2: DrawRLESpriteHelper2(image_type, 2); break;
+    case 3: DrawRLESpriteHelper2(image_type, 3); break;
+    default: assert(false); break;
+    }
+}
+
+#define DrawRLESpriteHelper1(image_type) \
+    DrawRLESprite1<image_type>(source_bits_pointer, dest_bits_pointer, palette_pointer, dpi, source_y_start, height, source_x_start, width)
+
+extern "C"
+{
+    /**
+     * Transfers readied images onto buffers
+     * This function copies the sprite data onto the screen
+     *  rct2: 0x0067AA18
+     */
+    void FASTCALL gfx_rle_sprite_to_buffer(const uint8* source_bits_pointer,
+                                             uint8* dest_bits_pointer,
+                                             const uint8* palette_pointer,
+                                             const rct_drawpixelinfo *dpi,
+                                             int image_type,
+                                             int source_y_start,
+                                             int height,
+                                             int source_x_start,
+                                             int width)
+    {
+        if (image_type & IMAGE_TYPE_USE_PALETTE)
+        {
+            if (image_type & IMAGE_TYPE_MIX_BACKGROUND)
+            {
+                DrawRLESpriteHelper1(IMAGE_TYPE_USE_PALETTE | IMAGE_TYPE_MIX_BACKGROUND);
+            }
+            else
+            {
+                DrawRLESpriteHelper1(IMAGE_TYPE_USE_PALETTE);
+            }
+        }
+        else if (image_type & IMAGE_TYPE_MIX_BACKGROUND)
+        {
+            DrawRLESpriteHelper1(IMAGE_TYPE_MIX_BACKGROUND);
+        }
+        else
+        {
+            DrawRLESpriteHelper1(0);
+        }
+    }
```

Comment:
wouldn't it suffice to call `helper1(image_type & (MIX_BACKGROUND | PALLETTED))` here? or does that confuse the compiler?